### PR TITLE
Inspector subscribe to event

### PR DESCRIPTION
**Pull request by jtsaito**
This PR adds support of the `subscribe_to_event` attribute to the `AWSInspectorAssessmentTemplate` resource. This feature is needed for automating AWS Inspector by infrastructure as code.

http://docs.aws.amazon.com/inspector/latest/APIReference/API_SubscribeToEvent.html

Follow up task:
- [ ] Add documentation if approved.

### CODE REVIEW

**Comment by jtsaito** [comment]
@jansiwy @fmasuhr ping

File: builtin/providers/aws/resource_aws_inspector_assessment_template.go [hunk 1]
```diff
@@ -119,3 +222,59 @@ func resourceAwsInspectorAssessmentTemplateDelete(d *schema.ResourceData, meta i
...
+	return
+}
+
+// onlyInFirstSubscribeToEventsMap return elements of 'a' which are not contained in 'b'
+func onlyInFirstSubscribeToEventsMap(a []map[string]interface{}, b []map[string]interface{}) (result []map[string]interface{}) {
+OUTER:
+	for _, elemA := range a {
+		for _, elemB := range b {
+			if elemA["event"] == elemB["event"] && elemA["topic_arn"] == elemB["topic_arn"] {
+				continue OUTER
```
Comment (jansiwy) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: sorry, but jumping to labels is IMO (even for golang) *too* ugly:

what extraction the inner loop into a separate function?

```go
func containsEventSubscription(s []map[string]interface{}, e map[string]interface{}) bool {
    for _, a := range s {
        if a["event"] == e["event"] && a["topic_arn"] == e["topic_arn"] {
            return true
        }
    }
    return false
}
```

File: builtin/providers/aws/resource_aws_inspector_assessment_template.go [hunk 2]
```diff
@@ -119,3 +222,59 @@ func resourceAwsInspectorAssessmentTemplateDelete(d *schema.ResourceData, meta i
...
+				results = append(results, m)
+			}
+		}
+	}
+
+	return
+}
+
+// onlyInFirstSubscribeToEventsMap return elements of 'a' which are not contained in 'b'
+func onlyInFirstSubscribeToEventsMap(a []map[string]interface{}, b []map[string]interface{}) (result []map[string]interface{}) {
```
Comment (jansiwy) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: the function name `onlyInFirstSubscribeToEventsMap` is IMO pretty quirky. actually it's set subscription implemented on slices: what about: `substractEventSubscriptions`?

File: builtin/providers/aws/resource_aws_inspector_assessment_template.go [hunk 3]
```diff
@@ -74,6 +101,23 @@ func resourceAwsInspectorAssessmentTemplateCreate(d *schema.ResourceData, meta i
 
 	d.SetId(*resp.AssessmentTemplateArn)
 
+	subscriptions := d.Get("subscribe_to_event").(*schema.Set)
+
+	for _, s := range subscriptions.List() {
+		m := s.(map[string]interface{})
+		event := m["event"].(string)
+		topicArn := m["topic_arn"].(string)
+		req, _ := conn.SubscribeToEventRequest(&inspector.SubscribeToEventInput{
```
Comment (jtsaito) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: wtf was i thinking?

File: builtin/providers/aws/resource_aws_inspector_assessment_template.go [hunk 4]
```diff
@@ -119,3 +222,59 @@ func resourceAwsInspectorAssessmentTemplateDelete(d *schema.ResourceData, meta i
 
 	return nil
 }
+
+// validateSubscribeToEvent validates the string is a known keyword
+func validateSubscribeToEvent(v interface{}, k string) (ws []string, errors []error) {
```
Comment (jtsaito) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: Yes, good point. This is strange, but supposed to be like this.

The signature of a validation function (`schema.SchemaValidateFunc`) requires a slice of strings to be returned for optional warnings.

File: builtin/providers/aws/resource_aws_inspector_assessment_template.go (hunk 2 above)
Comment (jtsaito) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: I was looking for a better name here and also considered something with "diff". Will stick with `substractEventSubscriptions`.

File: builtin/providers/aws/resource_aws_inspector_assessment_template.go (hunk 1 above)
Comment (jtsaito) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: But that would be ... more readable

Comment (jansiwy) [inline builtin/providers/aws/resource_aws_inspector_assessment_template.go]: @jtsaito i've already used one letter variables... isn't that enough unreadability? ;)

**Comment by jtsaito** [comment]
@jansiwy Thanks!

**Pull request closed without merging**